--- a/net/bridge/Kconfig
+++ b/net/bridge/Kconfig
@@ -4,7 +4,7 @@
 #
 
 config BRIDGE
-	tristate "802.1d Ethernet Bridging"
+	bool "802.1d Ethernet Bridging"
 	select LLC
 	select STP
 	depends on IPV6 || IPV6=n
--- a/net/8021q/Kconfig
+++ b/net/8021q/Kconfig
@@ -4,7 +4,7 @@
 #
 
 config VLAN_8021Q
-	tristate "802.1Q/802.1ad VLAN Support"
+	bool "802.1Q/802.1ad VLAN Support"
 	---help---
 	  Select this and you will be able to create 802.1Q VLAN interfaces
 	  on your Ethernet interfaces. 802.1Q VLAN supports almost
--- a/net/802/Kconfig
+++ b/net/802/Kconfig
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0-only
 config STP
-	tristate
+	bool
 	select LLC
 
 config GARP
--- a/net/llc/Kconfig
+++ b/net/llc/Kconfig
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0-only
 config LLC
-	tristate
+	bool
 	depends on NET
 
 config LLC2
--- a/net/packet/Kconfig
+++ b/net/packet/Kconfig
@@ -4,7 +4,7 @@
 #
 
 config PACKET
-	tristate "Packet socket"
+	bool "Packet socket"
 	---help---
 	  The Packet protocol is used by applications which communicate
 	  directly with network devices without an intermediate network
--- a/net/unix/Kconfig
+++ b/net/unix/Kconfig
@@ -4,7 +4,7 @@
 #
 
 config UNIX
-	tristate "Unix domain sockets"
+	bool "Unix domain sockets"
 	---help---
 	  If you say Y here, you will include support for Unix domain sockets;
 	  sockets are the standard Unix mechanism for establishing and
--- a/net/ipv4/Kconfig
+++ b/net/ipv4/Kconfig
@@ -471,7 +471,7 @@
 	See http://www.csc.ncsu.edu/faculty/rhee/export/bitcp/
 
 config TCP_CONG_CUBIC
-	tristate "CUBIC TCP"
+	bool "CUBIC TCP"
 	default y
 	---help---
 	This is version 2.0 of BIC-TCP which uses a cubic growth function
--- a/net/sched/Kconfig
+++ b/net/sched/Kconfig
@@ -310,7 +310,7 @@
 	  If unsure, say N.
 
 config NET_SCH_FQ_CODEL
-	tristate "Fair Queue Controlled Delay AQM (FQ_CODEL)"
+	bool "Fair Queue Controlled Delay AQM (FQ_CODEL)"
 	help
 	  Say Y here if you want to use the FQ Controlled Delay (FQ_CODEL)
 	  packet scheduling algorithm.
