--- a/switch_sdk/common/boardprofile/bprofile_YT9215SC_default_demo.c
+++ b/switch_sdk/common/boardprofile/bprofile_YT9215SC_default_demo.c
@@ -29,6 +29,8 @@
  *      Functions Declaration                      *
  **************************************************/
 yt_ret_t profile_yt9215sc_default_init(yt_hwProfile_info_t *hwprofile_info);
+extern uint32_t rd15_smi_read(uint8_t phy_addr, uint8_t reg_addr, uint16_t *pVal);
+extern uint32_t rd15_smi_write(uint8_t phy_addr, uint8_t reg_addr, uint16_t val);
 
 yt_swDescp_t yt9215sc_default_swDescp;
 
@@ -105,6 +107,12 @@
     yt9215sc_default_swDescp.chip_id = YT_SW_ID_9215;
     yt9215sc_default_swDescp.chip_model = YT_SW_MODEL_9215;
 
+    yt9215sc_default_swDescp.sw_access.swreg_acc_method = SWCHIP_ACC_SMI;
+    yt9215sc_default_swDescp.sw_access.controller.smi_controller.switchId = 0;
+    yt9215sc_default_swDescp.sw_access.controller.smi_controller.phyAddr = 0x1D;
+    yt9215sc_default_swDescp.sw_access.controller.smi_controller.smi_read = rd15_smi_read;
+    yt9215sc_default_swDescp.sw_access.controller.smi_controller.smi_write = rd15_smi_write;
+
     i = 0;
     while(YT9215sc2SGFibPortDescp[i].mac_id != INVALID_ID)
     {
